### ISSUE TITLE
The "Vault is unlocked" notification is not showing

### ISSUE THREAD

**Issue by alexbakker** [beemdevelopment/Aegis#1047]
I've completely disabled the "Vault is unlocked" notification for now, because I can't get it to work reliably without breaking other features.

For anyone who would like to look into this, make sure that things behave correctly in the following scenario's. Before every scenario, first unlock Aegis and make sure the notification appears:
- Kill the app using: ``adb shell am kill com.beemdevelopment.aegis.debug``
- Kill the app by swiping it away from the recent apps screen
- Lock the app by using all of the 3 auto locking triggers
- Lock the app using the "Lock" button
- Lock the app by tapping on the notification

After every scenario, check that the notification has disappeared. Then reopen the app and make sure it presents the lock screen and not the unlocked vault.

**Comment by alexbakker** [comment]
Here's what happens on my device (Pixel 5, Android 13) if I revert bb4450a00b33cce4527cfdbd4df04e709b690db4 and add the foreground service permission:
- Unlock Aegis
- (The notification appears)
- Minimize Aegis
- Swipe Aegis away from the recent apps list
- (The notification disappears)
- Start Aegis and notice that the vault is still unlocked

**Comment by alexbakker** [comment]
That's odd. Do you have auto lock on minimize enabled by any chance?. That'll make Aegis lock itself too early while reproducing the issue.

**Comment by siva-subramaniam-v** [comment]
Yes. I had all three auto-lock options enabled. I disabled them and the app behaves as you mentioned.

**Comment by siva-subramaniam-v** [comment]
I logged events in various files and this is what happened:

After launch:

---------------------------- PROCESS STARTED (24471) for package com.beemdevelopment.aegis.debug ----------------------------

Before unlocking vault:

1 -> provideVaultManager injects VaultManager
2 -> VaultManager() initialized
3 -> loadVaultFile() called
6 -> MainActivity onResume(). isRepoInitialized = false


After unlocking vault:

5 -> repo initialized => com.beemdevelopment.aegis.vault.VaultRepository@720e049
6 -> MainActivity onResume(). isRepoInitialized = true


After swiping away the app from recents:

- NotificationService Destroyed


After reopening the app:

6 -> MainActivity onResume(). isRepoInitialized = true


After swiping away the app from recent (once again):

---------------------------- PROCESS ENDED (24471) for package com.beemdevelopment.aegis.debug ----------------------------

![image](https://user-images.githubusercontent.com/112970189/234501573-58b98a59-5de2-4c69-aff4-aec57b28c894.png)

This indicates that the **process doesn't end** even after swiping it away from the recents screen. on reopening the repository remains initialized, ie: the _repo variable in VaultRepository remains initialized.

So the isVaultLoaded() method in VaultManager.java returns false

![image](https://user-images.githubusercontent.com/112970189/234503048-bb194916-14cb-4bee-b62b-99121ed02fe1.png)


on reopening the app, in onResume() of MainActivity, both the **if (!_vaultManager.isVaultLoaded())** and **else if (_loaded)** conditions fail

![image](https://user-images.githubusercontent.com/112970189/234503657-3cad98ee-9867-47d1-bcef-257654c0dd1e.png)

As the above image shows that the AuthScreen will be shown only **if (!_vaultManager.isVaultLoaded())** is true. I'm wondering why the process remains unkilled even after the app has been swiped away from the recents screen. Do you have any clue?

**Comment by siva-subramaniam-v** [comment]
@alexbakker I tried my best to find out why this happens, but couldn't find the root cause.

Hopefully, someone in the future will fix this.
